scribus/text/storytext.cpp

```diff
@@ -1671,25 +1674,14 @@ bool StoryText::selected(int pos) const
 int StoryText::selectWord(int pos)
 {
 	//Double click in a frame to select a word
-
-	int a = pos;
-	while(a > 0)
-	{
-		if (text(a-1).isLetterOrNumber())
-			--a;
-		else
-			break;
-	}
-	int b = pos;
-	while(b < length())
-	{
-		if (text(b).isLetterOrNumber())
-			++b;
-		else
-			break;
-	}
-	select(a, b - a);
-	return a;
+	if (m_WordBreakIterator == NULL)
+		m_WordBreakIterator = BreakIterator::createWordInstance(NULL, status);
+	m_WordBreakIterator->setText(plainText().utf16());
```

Comment:
You are still not checking `status` and using `m_WordBreakIterator` unconditionally, but `createWordInstance()` can fail.